ansible_builder/utils.py

```diff
@@ -1,8 +1,51 @@
 import subprocess
+import sys
+import os
+import yaml
 
+from . import constants
 
-def run_command(command):
+
+def run_command(command, capture_output=False):
     print('Running command:')
     print('  {0}'.format(' '.join(command)))
-    result = subprocess.run(command)
-    return bool(result.returncode == 0)
+
+    process = subprocess.Popen(command,
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.STDOUT)
+
+    output = []
+    for line in iter(process.stdout.readline, b''):
+        line = line.decode(sys.stdout.encoding)
+        if capture_output:
+            output.append(line.rstrip())
+        sys.stdout.write(line)
+
+    rc = process.poll()
+    return (rc, output)
+
+
+def introspect():
+    paths = []
+    path_root = os.path.join(constants.base_collections_path, 'ansible_collections')
+    if not os.path.exists(path_root):
+        # add debug statements at points like this
+        return paths
+    for namespace in sorted(os.listdir(path_root)):
+        for name in sorted(os.listdir(os.path.join(path_root, namespace))):
```

Comment:
There should be a check before this that the location is a directory, otherwise `os.listdir` can error being given a file.